### PR TITLE
Possible solution for testing actors

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: shoebox/test/com/keepit/classify/DomainClassifierTest.scala [hunk 1]
```diff
@@ -13,28 +13,35 @@ import com.keepit.common.net.{HttpClient, FakeHttpClient}
...
+import akka.actor.ActorSystem
+import akka.testkit.TestKit
 import play.api.Play.current
 import play.api.test.Helpers.running
 
-class DomainClassifierTest extends Specification with DbRepos {
+class DomainClassifierTest extends TestKit(ActorSystem()) with Specification with DbRepos {
+
+  private def await[T](f: Future[T]): T = {
+    Await.result(f, pairIntToDuration(100, TimeUnit.SECONDS))
```
Comment (andrewconner) [inline shoebox/test/com/keepit/classify/DomainClassifierTest.scala]: Should we (potentially) wait this long? Ideally, 1 second (or less?) should fit pretty much all cases — in fact, anything taking longer than a few seconds is likely a bad test.

You could make it accept an implicit `Duration`, and then have a default of x00 ms.

File: shoebox/app/com/keepit/common/actor/ActorFactory.scala [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
-
+@ImplementedBy(classOf[ActorBuilderImpl])
+trait ActorBuilder {
+  def apply(system: ActorSystem, provider: Provider[_ <: FortyTwoActor]): ActorRef
+}
+class ActorBuilderImpl extends ActorBuilder {
+  def apply(system: ActorSystem, provider: Provider[_ <: FortyTwoActor]): ActorRef =
+    system.actorOf(Props { provider.get })
+}
+class TestActorBuilderImpl extends ActorBuilder {
```
Comment (eishay) [inline shoebox/app/com/keepit/common/actor/ActorFactory.scala]: Lets not have tests artifacts with the production code, test artifacts should be under shoebox/test (like with the FakeFoo classes)

File: shoebox/app/com/keepit/common/actor/ActorFactory.scala [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+  def apply(system: ActorSystem, provider: Provider[_ <: FortyTwoActor]): ActorRef =
+    TestActorRef(provider.get)(system)
```
Comment (eishay) [inline shoebox/app/com/keepit/common/actor/ActorFactory.scala]: nice!

File: shoebox/project/Build.scala [hunk 4]
```diff
@@ -36,7 +36,7 @@ object ApplicationBuild extends Build {
       jdbc,
       "mysql" % "mysql-connector-java" % "5.1.10",
       "org.clapper" %% "grizzled-slf4j" % "1.0.1",
-      "com.typesafe.akka" % "akka-testkit" % "2.0.2",
+      "com.typesafe.akka" %% "akka-testkit" % "2.1.0",
```
Comment (eishay) [inline shoebox/project/Build.scala:39]: what's the difference between % and %% ?

File: shoebox/test/com/keepit/classify/DomainClassifierTest.scala (hunk 1 above)
Comment (gmethvin) [inline shoebox/test/com/keepit/classify/DomainClassifierTest.scala]: I actually meant to make this 100ms, not 100s. I think that should be enough most of the time, right?

**Comment by eishay** [comment]
nice, lets see how it works

File: shoebox/project/Build.scala (hunk 4 above)
Comment (andrewconner) [inline shoebox/project/Build.scala:39]: It tells sbt that the library is cross-built, and to find the version for whatever version of Scala we're using.

**Comment by andrewconner** [comment]
Really great.

File: shoebox/app/com/keepit/common/actor/ActorFactory.scala (hunk 2 above)
Comment (gmethvin) [inline shoebox/app/com/keepit/common/actor/ActorFactory.scala]: I agree.

File: shoebox/project/Build.scala (hunk 4 above)
Comment (eishay) [inline shoebox/project/Build.scala:39]: oh, thanks

**Pull request merged**